doc/changelog.md

```diff
@@ -46,114 +139,169 @@
-  [(#375)](https://github.com/PennyLaneAI/catalyst/pull/375)
+  ```python
+  dev = qml.device("lightning.qubit", wires=2, shots=1000, mcmc=True)
 
-* ``AllocOp``, ``DeallocOp`` have now (only) value semantics. In the frontend, the last
-  quantum register is deallocated instead of the first one. This allows to return the quantum
-  register in functions and can be given to another function (useful for quantum transformation).
-  [(#360)](https://github.com/PennyLaneAI/catalyst/pull/360)
+  @qml.qnode(dev)
+  def circuit(x):
+      qml.RX(x, wires=0)
+      return qml.expval(qml.PauliZ(0))
+  ```
+
+  ```pycon
+  >>> circuit(0.54)
+  array(0.856)
+  ```
+
+* Improvements have been made to the runtime and quantum MLIR dialect in order
+  to support asynchronous execution.
+
+  - The runtime now supports multiple active devices managed via a device pool. The new `RTDevice`
+    data-class and `RTDeviceStatus` along with the `thread_local` device instance pointer enable
+    the runtime to better scope the lifetime of device instances concurrently. With these changes,
+    one can create multiple active devices and execute multiple programs in a multithreaded
+    environment.
+    [(#381)](https://github.com/PennyLaneAI/catalyst/pull/381)
+
+  - The ability to dynamically release devices has been added via `DeviceReleaseOp` in the Quantum
+    MLIR dialect. This is lowered to the `__quantum__rt__device_release()` runtime instruction,
+    which updates the status of the device instance from `Active` to `Inactive`. The runtime will reuse
+    this deactivated instance instead of creating a new one automatically at runtime in a
+    multi-QNode workflow when another device with identical specifications is requested.
+    [(#381)](https://github.com/PennyLaneAI/catalyst/pull/381)
+
+  - The `DeviceOp` definition in the Quantum MLIR dialect has been updated to lower a tuple
+    of device information `('lib', 'name', 'kwargs')` to a single device initialization call
+    `__quantum__rt__device_init(int8_t *, int8_t *, int8_t *)`. This allows the runtime to initialize
+    device instances without keeping partial information of the device
+    [(#396)](https://github.com/PennyLaneAI/catalyst/pull/396)
 
 * The quantum adjoint compiler routine has been extended to support function calls that affect the
   quantum state within an adjoint region. Note that the function may only provide a single result
-  consisting of the quantum register.
-  By itself this provides no user-facing changes, but compiler pass developers may now generate
-  quantum adjoint operations around a block of code containing function calls as well as quantum
-  operations and control flow operations.
+  consisting of the quantum register. By itself this provides no user-facing changes, but compiler
+  pass developers may now generate quantum adjoint operations around a block of code containing
+  function calls as well as quantum operations and control flow operations.
```

Comment:
@rmoyard could you explain this change to me as well?

Reply:
You can use have `func.call` inside `quantum.adjoint` operation and it will work.

```
func.func circuit(...)

quantum.adjoint(qreg) {
    func.call(circuit)
}
```